solr/core/src/java/org/apache/solr/cloud/RecoveryStrategy.java

```diff
@@ -223,8 +223,10 @@ private final void replicate(String nodeName, SolrCore core, ZkNodeProps leaderp
 
     log.info("Attempting to replicate from [{}].", leaderUrl);
 
-    // send commit
-    commitOnLeader(leaderUrl);
+    // send commit if replica could be a leader
+    if (replicaType == null || Replica.Type.isLeaderType(replicaType)) {
```

Comment:
minor: `Replica.Type.isLeaderType` could also check for null and make this logic clearer